src/trajectory.jl

```diff
@@ -57,7 +58,9 @@ function transition(
     if is_accept
         z = PhasePoint(z′.θ, -z′.r, z′.ℓπ, z′.ℓκ)
     end
-    return z, α
+    stattuple = (τ.integrator.ϵ, τ.n_steps, is_accept, α, z.ℓπ.value, -neg_energy(z))
+    statnames = (:step_size, :n_steps, :is_accept, :acceptance_rate, :log_density, :hamiltonian_energy)
```

Comment:
Can we replace the following

```julia

stattuple = (τ.integrator.ϵ, τ.n_steps, is_accept, α, z.ℓπ.value, -neg_energy(z))
    statnames = (:step_size, :n_steps, :is_accept, :acceptance_rate, :log_density, :hamiltonian_energy)
```

with

```julia

stats = (step_size=.integrator.ϵ, 
              n_steps=τ.n_steps, 
              is_accept=is_accept, 
              acceptance_rate=α, 
              log_density=z.ℓπ.value, 
              hamiltonian_energy=-neg_energy(z))
```

I find the later a bit easier to read and maintain (say add a new field to stats).

Reply:
Yes definitely. I was originall try to grab the type information explicitly but ended finding this doesn't matter by running `@code_warntype`.